crd/apis/network/v1/conditions.go

```diff
@@ -0,0 +1,124 @@
+type GKENetworkParamSetConditionType string
+
+const (
+	// GKENetworkParamSetStatusReady is the condition type that holds
+	// if the GKENetworkParamSet object is validated
+	GKENetworkParamSetStatusReady GKENetworkParamSetConditionType = "Ready"
+)
+
+// GKENetworkParamSetConditionReason defines the set of reasons that explain why a
+// particular GKENetworkParamSet condition type has been raised.
+type GKENetworkParamSetConditionReason string
+
+const (
+	// SubnetNotFound indicates that the specified subnet was not found.
+	SubnetNotFound GKENetworkParamSetConditionReason = "SubnetNotFound"
+	// SecondaryRangeNotFound indicates that the specified secondary range was not found.
+	SecondaryRangeNotFound GKENetworkParamSetConditionReason = "SecondaryRangeNotFound"
+	// DeviceModeCantBeUsedWithSecondaryRange indicates that device mode was used with a secondary range.
+	DeviceModeCantBeUsedWithSecondaryRange GKENetworkParamSetConditionReason = "DeviceModeCantBeUsedWithSecondaryRange"
+	// DeviceModeVPCAlreadyInUse indicates that the VPC is already in use by another GKENetworkParamSet resource.
+	DeviceModeVPCAlreadyInUse GKENetworkParamSetConditionReason = "DeviceModeVPCAlreadyInUse"
+	// DeviceModeCantUseDefaultVPC indicates that a device mode GKENetworkParamSet cannot use the default VPC.
+	DeviceModeCantUseDefaultVPC GKENetworkParamSetConditionReason = "DeviceModeCantUseDefaultVPC"
+	// DPDKUnsupported indicates that DPDK device mode is not supported on the current cluster.
+	DPDKUnsupported GKENetworkParamSetConditionReason = "DPDKUnsupported"
+)
+
+// GNPNetworkParamsReadyConditionReason defines the set of reasons that explains
+// the ParamsReady condition on the referencing Network resource.
+type GNPNetworkParamsReadyConditionReason string
+
+const (
+	// L3SecondaryMissing indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with secondary range unspecified.
+	L3SecondaryMissing GNPNetworkParamsReadyConditionReason = "L3SecondaryMissing"
+	// L3DeviceModeExists indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with device mode specified.
+	L3DeviceModeExists GNPNetworkParamsReadyConditionReason = "L3DeviceModeExists"
+	// DeviceModeMissing indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with device mode unspecified.
+	DeviceModeMissing GNPNetworkParamsReadyConditionReason = "DeviceModeMissing"
+	// DeviceSecondaryExists indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with a secondary range specified.
+	DeviceSecondaryExists GNPNetworkParamsReadyConditionReason = "DeviceSecondaryExists"
+)
+
+// ###############
+// ### helpers ###
+
+// GetCondition returns the "condType" condition from the obj. Only Network and
+// GKENetworkParamSet types are supported. Returns nil when not found.
+func GetCondition(obj interface{}, condType string) (*metav1.Condition, error) {
```

Comment:
suggestion

create a struct that wraps metav1.Condition that implements the specific CRUD for conditions that we care about. Might be cleaner for users than one-off functions.

```
func NewConditionCRUD(c*metav1.Condition) ...
type ConditionCRUD struct { ... }


crud := NewConditionCRUD(obj.Status.Conditions)
if crud.Ready() { ...
}

crud.SetReady(false, "it's not ready")
```

Reply:
As I were looking into this I found https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go which has all the required functions, will close this PR.